cmd/internal/cli/startvm_darwin.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package cli
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/sylabs/singularity/internal/pkg/buildcfg"
+	"github.com/sylabs/singularity/internal/pkg/sylog"
+)
+
+func startVM(sifImage, singAction, cliExtra string, isInternal bool) error {
+	// Setup some needed variables
+	hdString := fmt.Sprintf("2:0,ahci-hd,%s", sifImage)
+	bzImage := fmt.Sprintf(buildcfg.LIBEXECDIR+"%s"+runtime.GOARCH, "/singularity/vm/syos-kernel-")
+	initramfs := fmt.Sprintf(buildcfg.LIBEXECDIR+"%s"+runtime.GOARCH+".gz", "/singularity/vm/initramfs_")
+
+	// Default xhyve Arguments
+	defArgs := []string{"-A", "-m", VMRAM, "-c", VMCPU, "-s", "0:0,hostbridge", "-s", hdString, "-s", "31,lpc", "-l", "com1,stdio"}
+
+	if len(BindPaths) > 8 {
+		sylog.Fatalf("Maximum of 8 bind mounts")
+	}
+
+	// Bind mounts
+	singBinds := []string{""}
+
+	slot := 5
+
+	for idx, bindpath := range BindPaths {
+		splitted := strings.Split(bindpath, ":")
+		src := splitted[0]
+		dst := ""
+		if len(splitted) > 1 {
+			dst = splitted[1]
+		} else {
+			dst = src
+		}
+
+		sylog.Debugf("Bind path: " + src + " -> " + dst)
+		// TODO: Figure out if src is a directory or not
+		mntTag := filepath.Base(src)
+
+		pciArgs := fmt.Sprintf("%s:%s,virtio-9p,%s=%s", strconv.Itoa(slot), strconv.Itoa(idx), mntTag, src)
+		defArgs = append(defArgs, "-s", pciArgs)
+
+		localBind := fmt.Sprintf("%s:%s", mntTag, dst)
+		singBinds = append(singBinds, localBind)
+
+		sylog.Debugf("PCI: %s", pciArgs)
+	}
+
+	usr, err := user.Current()
+	if err != nil {
+		sylog.Fatalf("Failed to get current user")
+	}
+
+	// Force $HOME to be mounted
+	// TODO: engineConfig.GetHomeSource() / GetHomeDest() -- should probably be used
+	homeSrc := usr.HomeDir
+	pciArgs := fmt.Sprintf("4:0,virtio-9p,home=%s", homeSrc)
+	homeBind := fmt.Sprintf("home:%s", homeSrc)
+	singBinds = append(singBinds, homeBind)
+
+	sylog.Debugf("PCI: %s", pciArgs)
+	defArgs = append(defArgs, "-s", pciArgs)
+
+	userInfo := fmt.Sprintf("%s:%s:%s", usr.Username, usr.Uid, usr.Gid)
+
+	if IsSyOS {
+		cliExtra = "syos"
```

Comment:
I want to make sure this is intentional: this is discarding cliExtra passed as an argument to startVM and replacing it with syos. If that's the case, a comment would help.

Reply:
Yup ... You can replicate the `--syos` flag with:  `singularity exec [image] syos` ...  We're ignoring anything passed since we want a SyOS shell ... We aren't going into the image automatically here.